src/BSON/Regex.c

```diff
@@ -206,6 +207,24 @@ PHP_METHOD(Regex, __toString)
 }
 /* }}} */
 
+/* {{{ proto array Regex::jsonSerialize()
+*/
+PHP_METHOD(Regex, jsonSerialize)
+{
+	php_phongo_regex_t *intern;
+
+	if (zend_parse_parameters_none() == FAILURE) {
+		return;
+	}
+
+	intern = Z_REGEX_OBJ_P(getThis());
+
+	array_init_size(return_value, 2);
+	ADD_ASSOC_STRINGL(return_value, "$regex", intern->pattern, intern->pattern_len);
+	ADD_ASSOC_STRINGL(return_value, "$options", intern->flags, intern->flags_len);
```

Comment:
I think that we should order the flags in the constructor/initialiser only. That should be all what we need to do about this I believe.

The HHVM ticket is [HHVM-264](https://jira.mongodb.org/browse/HHVM-264).